### PR TITLE
Add ES6 Modules

### PR DESCRIPTION
**Pull request by iamsjunaid**
In this milestone, I've added ES6 modules and functionalities into the awesome books project.

### CODE REVIEW

File: modules/storage.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+export function newAuthorData() {
+  const authorData = {
+    bookName: bookName.value,
+    bookAuthor: bookAuthor.value,
+    bookNumber: Math.floor(Math.random() * 1000000),
+  };
+  storedData.push(authorData);
+  localStorage.setItem('storedData', JSON.stringify(storedData));
+}
```
Comment (GedeonTS) [inline modules/storage.js]: - According to [this requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/books/books_with_es6.md#:~:text=Refactor%20your%20functions%20to%20arrow%20functions.), you are supposed to refactor your functions to arrow functions to make your code more concise, readable and maintainable

File: modules/storage.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+export function deleteBook(bookNumber) {
+  const updatedData = storedData.filter((storedData) => storedData.bookNumber !== bookNumber);
+  localStorage.setItem('storedData', JSON.stringify(updatedData));
+}
```
Comment (GedeonTS) [inline modules/storage.js]: - According to [this requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/books/books_with_es6.md#:~:text=Refactor%20your%20functions%20to%20arrow%20functions.), you are supposed to refactor your functions to arrow functions to make your code more concise, readable and maintainable

File: modules/storage.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+export function getStoredData() {
+  return storedData;
+}
```
Comment (GedeonTS) [inline modules/storage.js]: - According to [this requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/books/books_with_es6.md#:~:text=Refactor%20your%20functions%20to%20arrow%20functions.), you are supposed to refactor your functions to arrow functions to make your code more concise, readable and maintainable

File: index.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { form, bookName, bookAuthor } from './modules/form.js';
+import { newAuthorData } from './modules/storage.js';
+import { showBookList } from './modules/bookList.js';
+
```
Comment (GedeonTS) [inline index.js:6]: - According to [this requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/books/books_with_es6.md#:~:text=Display%20the%20current,your%20future%20job.), you are supposed to display the current date below the navigation bar using [Luxon](https://moment.github.io/luxon/#/install?id=es6) Library. Please Kindly add that to your project.

Screenshot from 2023-05-01 16-29-50

**Review by GedeonTS — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED :recycle: :

Hi @iamsjunaid , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :tophat:  :+1: 
- Linters are Passing :+1: 
- Meaningful Commit Messages :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by gemmen29 — APPROVED** [review]
Hi @iamsjunaid, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**